CHANGES.md

```diff
@@ -258,6 +258,26 @@ OpenSSL 3.1
 
    *Tomáš Mráz*
 
+ * Change the default salt length for PKCS#1 RSASSA-PSS signatures to the
+   maximum size that is smaller or equal to the digest length to comply with
+   FIPS 186-4 section 5. This is implemented by a new option
+   `OSSL_PKEY_RSA_PSS_SALT_LEN_AUTO_DIGEST_MAX` ("auto-digestmax") for the
+   `rsa_pss_saltlen` parameter, which is now the default. Signature
+   verification is not affected by this change and continues to work as before.
+
+   In practice, this means that many common combinations of RSA key size and
+   digest algorithm will now use shorter salts. J.-S. Coron shows in "Optimal
+   Security Proofs for PSS and Other Signature Schemes. Advances in Cryptology
+   – Eurocrypt 2002, volume 2332 of Lecture Notes in Computer Science, pp. 272
+   – 287. Springer Verlag, 2002." that longer salts than the output size of
+   modern hash functions do not increase security: "For example, for an
+   application in which at most one billion signatures will be generated,
+   k0 = 30 bits of random salt are actually sufficient to guarantee the same
+   level of security as RSA, and taking a larger salt does not increase the
+   security level."
+
```

Comment:
Please drop this paragraph. It suffices for this information to be in the commit message.